src/shell/shell_cmds.cpp

```diff
@@ -472,19 +472,24 @@ static std::string format_number(size_t num)
 	const auto tb = static_cast<unsigned>(num);
 	char buf[22];
 	if (tb) {
-		safe_sprintf(buf, "%u,%03u,%03u,%03u,%03u", tb, gb, mb, kb, b);
+		safe_sprintf(buf, "%u%c%03u%c%03u%c%03u%c%03u", tb,
+		             dos.tables.country[7], gb, dos.tables.country[7], mb,
+		             dos.tables.country[7], kb, dos.tables.country[7], b);
 		return buf;
 	}
 	if (gb) {
-		safe_sprintf(buf, "%u,%03u,%03u,%03u", gb, mb, kb, b);
+		safe_sprintf(buf, "%u%c%03u%c%03u%c%03u", gb,
+		             dos.tables.country[7], mb, dos.tables.country[7],
+		             kb, dos.tables.country[7], b);
 		return buf;
 	}
 	if (mb) {
-		safe_sprintf(buf, "%u,%03u,%03u", mb, kb, b);
+		safe_sprintf(buf, "%u%c%03u%c%03u", mb, dos.tables.country[7],
+		             kb, dos.tables.country[7], b);
 		return buf;
 	}
 	if (kb) {
-		safe_sprintf(buf, "%u,%03u", kb, b);
+		safe_sprintf(buf, "%u%c%03u", kb, dos.tables.country[7], b);
```

Comment:
There are ten uses of `dos.tables.country[7]` here (and.. to a casual reader, the `dos.tables.country[7]` says "This is the 7th country".

This is just one examples, but all the magic numbers need to lifted up to constexpr's (or type-sized enums, when the values all fall into the same genre or category).  Or some other self-documenting structure.

Reply:
I added new variables like thousands_separator, date_separator, and time_separator pointing to the chars in the country structure so that the code readability should have been improved.